CHANGELOG.md

```diff
@@ -2,9 +2,62 @@
 
 ## [Unreleased]
 
-### Fixed
+### Changed
 
+- Also apply `empty_item_single_line=true` to trait definitions [#5047](https://github.com/rust-lang/rustfmt/issues/5047)
+- Replace `structopt` dependency with `clap` [PR #5239](https://github.com/rust-lang/rustfmt/pull/5239)
```

Comment:
In my opinion this is still an internally facing change that's not of particular interest for end users. I suspect part of the friction here is trying to balance the benefits of calling out the work that's been contributed vs. a minimal/succinct/clear/etc. summary of changes relevant for end users.

I don't want to add any additional overhead on ourselves given how we've struggled to maintain this already, but wonder if having a separate changelog (broader content included) vs. release notes (strictly end user focused) would help? 

wdyt @ytmimi?